os/pm/pm_initialize.c

```diff
@@ -73,6 +74,242 @@
+	/* Case when all state threshold values are provided by PM_STATE_ENTER_THRESH*/
+
+		assign_defaultvalue = false;
+
+	} else if((check_count > 0 && check_count < (CONFIG_PM_NSTATE - 1))) {
+	/* Case when less state threshold values are provided by PM_STATE_ENTER_THRESH*/
+
+		DEBUGASSERT(!(check_count > 0 && check_count < CONFIG_PM_NSTATE));
+
+	} else {
+	/* Case when no state threshold values are provided by PM_STATE_ENTER_THRESH*/
+
+		assign_defaultvalue = true;
+
+	}
+#endif
+
+	/* If assign_defaultvalue is true (any of the state enter threshold values are not provided),
+	 * give default threshold values (in steps of 10).
+	 */
+
+	if(assign_defaultvalue) {
+
+		for(index = 1; index < CONFIG_PM_NSTATE; index++) {
+			pm_sleep_enter_activity_thresholds[index] = PM_DEFAULT_STATE_ENTER_THRESH;
+		}
+	}
+
+	/* Reset flag to assign default threshold values */
+	assign_defaultvalue = true;
+
+#ifdef CONFIG_PM_STATE_EXIT_THRESH
+
+	/* Initialize state threshold value to be read from thresh_exit */
+	int threshold_exit = 0;
+	const char *thresh_exit = CONFIG_PM_STATE_EXIT_THRESH;
+
+	/* Variable to track number of activity threshold provided by the SOC */
+	check_count = 0;
+
+	/* Normal state entering threshold is NA, hence initialized to 0 */
+	index = 0;
+	pm_sleep_enter_activity_thresholds[index++] = 0;
+
+	/* Parse the sleep state enter thresholds from the Kconfig */
+
+	while(*thresh_exit) {
+
+		/* Extract thresholds from string input in config as integer value */
+		threshold_exit = strtoul(thresh_exit, NULL, 0);
+
+		while(*thresh_exit != ',' && *thresh_exit) {
+			thresh_exit++;
+		}
+
+		if(*thresh_exit == ',') {
+			thresh_exit++;
+		}
+
+		check_count++;
+
+		/* Case when more state threshold values are provided by PM_STATE_EXIT_THRESH*/
+		if(check_count > (CONFIG_PM_NSTATE - 1)) {
+			DEBUGASSERT(!(check_count > (CONFIG_PM_NSTATE - 1)));
+		}
+
+		/* Update sleep state threshold array with extracted integer value */
+		pm_sleep_exit_activity_thresholds[index++] = threshold_exit;
+	}
+
+	/* Check if all the state entering tick count thresholds are provided by the SOC */
+
+	if(check_count == (CONFIG_PM_NSTATE - 1)) {
+	/* Case when all state threshold values are provided by PM_STATE_EXIT_THRESH*/
+
+		assign_defaultvalue = false;
+
+	} else if((check_count > 0 && check_count < (CONFIG_PM_NSTATE - 1))) {
+	/* Case when less state threshold values are provided by PM_STATE_EXIT_THRESH*/
+
+		DEBUGASSERT(!(check_count > 0 && check_count < CONFIG_PM_NSTATE));
+
+	} else {
+	/* Case when no state threshold values are provided by PM_STATE_EXIT_THRESH*/
+
+		assign_defaultvalue = true;
+
+	}
+#endif
+
+	/* If assign_defaultvalue is true (any of the state exit threshold values are not provided),
+	 * give default threshold values (in steps of 10).
+	 */
+
+	if(assign_defaultvalue) {
+
+		for(index = 1; index < CONFIG_PM_NSTATE; index++) {
+			pm_sleep_exit_activity_thresholds[index] = PM_DEFAULT_STATE_EXIT_THRESH;
+		}
+	}
+}
+
```

Comment:
Code to handle SWITCH, ENTER and EXIT is 100% repetitive, please consolidate to reduce to single function and pass the deviating variables as parameters. There should be 1 instance of code that shall be called 3 times in ur case